spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/domain/EntityScanner.java

```diff
@@ -42,13 +42,30 @@
 
 	private final ApplicationContext context;
 
+	private final ClassPathScanningCandidateComponentProvider componentProvider;
+
 	/**
 	 * Create a new {@link EntityScanner} instance.
 	 * @param context the source application context
 	 */
 	public EntityScanner(ApplicationContext context) {
 		Assert.notNull(context, "Context must not be null");
 		this.context = context;
+		this.componentProvider = new ClassPathScanningCandidateComponentProvider(false);
+		this.componentProvider.setEnvironment(this.context.getEnvironment());
+		this.componentProvider.setResourceLoader(this.context);
+	}
+
+	/**
+	 * Create a new {@link EntityScanner} instance.
+	 * @param context the source application context
+	 * @param componentProvider the scanner, it's environment and resource loader must
+	 * already be set
+	 */
+	public EntityScanner(ApplicationContext context, ClassPathScanningCandidateComponentProvider componentProvider) {
```

Comment:
I mentioned in my review "I would expose a protected method where the ClassPathScanningCandidateComponentProvider can be tuned.". 

There is nothing wrong having to extend from `EntityScanner` for your use case. I don't think exposing the provider as a parameter is warranted for your use case either.